core/src/main/java/org/springframework/security/provisioning/InMemoryUserDetailsManager.java

```diff
@@ -143,7 +143,7 @@ public void changePassword(String oldPassword, String newPassword) {
 	@Override
 	public UserDetails updatePassword(UserDetails user, String newPassword) {
 		String username = user.getUsername();
```

Comment:
This is missing check `Assert.isTrue(userExists(user.getUsername()), "user should exist");`